dask/array/reductions.py

```diff
@@ -383,7 +383,9 @@ def partial_reduce(
     if np.isscalar(meta):
         return Array(graph, name, out_chunks, dtype=dtype)
     else:
-        with contextlib.suppress(AttributeError):
+        with contextlib.suppress(AttributeError), warnings.catch_warnings():
+            if name.startswith("var") or name.startswith("moment"):
+                warnings.simplefilter("ignore", np.ComplexWarning)
```

Comment:
Does this it make sense to only filter warnings here if `meta` is a complex dtype?

Reply:
This would suppress a `ComplexWarning` on all operations where downcasting occurs, wouldn't it? I wanted to replicate the behaviour of numpy, which raises such a warning when we call `var` on a complex array, but specify a real `dtype` as output:

- `np.var(a)` - no complex warning in numpy, but without this if clause we would get a warning in dask, due to output `dtype` being a float
- `np.var(a, dtype='c8')` - no warning in numpy and dask
- `np.var(a, dtype='f4')` - we should get a warning in both numpy and dask here

All of these cases are covered by unit tests. I know this block is not elegant, but a better one would probably require going deeper into Dask internals and I want to keep the changes minimal, considering it is a minor issue and I'm new to Dask.

Reply:
I was thinking of keeping the check on the aggregation `name` and including a check for `meta` being complex, so something like

```suggestion
            if name.startswith("var") or name.startswith("moment") and np.iscomplexobj(meta):
                warnings.simplefilter("ignore", np.ComplexWarning)
```

With the thought being that we would want this warning suppressed only in the case where we know downcasting would occur during a `var` or `moment` operation - does that make sense, or am I missing something?